README.md

```diff
@@ -0,0 +1,65 @@
+# DistributedCircuitBreaker.NET
+
+DistributedCircuitBreaker.NET is a Circuit Breaker that allows developers store data and state of the circuit using an external repository such as SqlServer, Redis, MongoDB Etc. by providing a interface. Redis implementation is included.
+
+# How to Use
+
+The simplest way to use it:
+* Add the dependency to IServiceCollection (located usually in Startup class)
+```csharp
+            services.AddDistributedRedisCircuitBreaker(options =>
+            {
+                options.DurationOfBreak = TimeSpan.FromSeconds(300);
+                options.WindowDuration = TimeSpan.FromSeconds(60);
+                options.RedisConnectionConfiguration = "localhost:6379";
+            });
+```
+
+* Get an instance of CircuitBreaker using the `ICircuitBreakerFactory` interface
+* Call Create method passing:
+1. A key used as a unique identifier to perform the operations and compute failures and success
+2. Minimum number of failures before starting considering counters
+3. Percent of failures that should cause CB be opened
+
+** Both 2 and 3 rules should be broken to make the circuit to be opened
+
+```csharp
+
+public class CatalogService : ICatalogService
+{
+    private readonly ICircuitBreakerFactory _circuitBreakerFactory;
+
+    public CatalogService(ICircuitBreakerFactory circuitBreakerFactory)
+    {
+        _circuitBreakerFactory = circuitBreakerFactory;
+    }
+
+    public async Task<Catalog> GetCatalogItems(int page, int take, 
+                                               int? brand, int? type)
+    {
+        var uri = API.Catalog.GetAllCatalogItems(_remoteServiceBaseUrl, 
+                                                 page, take, brand, type);
+
+	int exceptionsAllowedBeforeBreaking = 20;
+	decimal failureRateAllowed = 0.5M;
+
+	var circuitBreaker = _circuitBreakerFactory.Create(uri.AbsoluteUri, exceptionsAllowedBeforeBreaking , failureRateAllowed);
+
+	var responseString = await circuitBreaker.ExecuteActionAsync(() =>  _httpClient.GetStringAsync(uri)); 
+
+        var catalog = JsonConvert.DeserializeObject<Catalog>(responseString);
+        return catalog;
+    }
+}
+
+```
+
+# Nuget Packages
+
+ ## DistributedCircuitBreaker
+
+ This is the Circuit Breaker engine. It depends on a implementation of a public repository interface `IDistributedCircuitBreakerRepository`
+
+ ## DistributedCircuitBreakerRedis
+
+ It is the implementation of the interface `IDistributedCircuitBreakerRepository` using a Redis as repository.	 
```

Comment:
It would be nice add some documentation about CB exceptions and how to treat that.

Reply:
Nice. I'll add it